torch/distributed/distributed_c10d.py

```diff
@@ -75,8 +76,22 @@ def __new__(cls, name):
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
             raise ValueError("Invalid backend: '{}'".format(name))
+        elif value != Backend.GLOO and value != Backend.NCCL and value != Backend.MPI:
+            value = name
         return value
 
+    @classmethod
+    def register_backend(cls, name, func):
+        """Registers a new backend.
```

Comment:
Does it show properly in the doc? Could you please try the following and paste the screenshot here? Thanks!

```
cd docs/
pip install -r requirements.txt
make html
```

You can check the result by using this [gist](https://gist.github.com/colesbury/df218a96334dbd31ee0e93760f70f114) to serve your docs and paste a screenshot in this PR.

Reply:
@mrshenli, make html would fail with the below error.

FileNotFoundError: [Errno 2] No such file or directory: 'katex': 'katex'

but lucky thing is distributed.html has been generated. I paste the screenshot for your reference.
https://user-images.githubusercontent.com/16394660/76596980-70036280-653a-11ea-8d57-f36ee54894c2.png

![image](https://user-images.githubusercontent.com/16394660/76596980-70036280-653a-11ea-8d57-f36ee54894c2.png)